### PR TITLE
Feature/settings activity

### PR DESCRIPTION
**Pull request by BartSoftware**

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java [hunk 1]
```diff
@@ -46,6 +50,25 @@ public void prepareView() {
         extractBundle();
     }
 
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.settings:
+                Intent intent = new Intent(this, SettingsActivity.class);
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: Please move building intent logic into SettingsActivity static method.
Furthermore, I would move all starting activity (Intent) code to separate method.

Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: please create intent from internal static method in SettingsActivity class (like EditActivity class)

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsViewModel.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+
+import android.databinding.ObservableBoolean;
+
+public class SettingsViewModel {
+
+    private final ObservableBoolean isEnabled = new ObservableBoolean(false);
+
+    public ObservableBoolean getIsEnabled() {
+        return isEnabled;
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsViewModel.java]: what is enabled? I would prefer to change this name to more descriptive about this case e.g. isClearServiceEnabled. In the future, we can have more settings so it couldn't be obvious.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setTitle(getString(R.string.settings));
+        }
+    }
+
+    private void prepareViews() {
+        SettingsActivityBinding binding =  DataBindingUtil.setContentView(this, R.layout.settings_activity);
+        SettingsViewModel viewModel = new SettingsViewModel();
+        binding.setViewModel(viewModel);
+        binding.periodicClearingSwitch.setOnCheckedChangeListener(
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java]: I would like to see settings checked listener inside viewModel

File: app/src/main/res/layout/settings_activity.xml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                android:layout_weight="3"
+                android:text="@string/enable_periodic_clearing"
+                android:textSize="@dimen/section_description_text_size"/>
+
+            <Switch
+                android:id="@+id/periodic_clearing_switch"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center"
+                android:checked="@{viewModel.getIsEnabled().get()}"
```
Comment (michal1990) [inline app/src/main/res/layout/settings_activity.xml]: I believe simple viewModel.isEnabled will be enough. Data binding automatically finds out getter for this field if exists and collects a value from Observable

File: app/src/main/res/layout/settings_activity.xml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            android:weightSum="4">
+
+            <TextView
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_margin="@dimen/text_view_margin"
+                android:layout_weight="3"
+                android:enabled="@{viewModel.getIsEnabled().get()}"
```
Comment (michal1990) [inline app/src/main/res/layout/settings_activity.xml]: Did you try to disable the whole view? (not it's child)

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -15,6 +15,12 @@
     <string name="save">Save</string>
     <string name="clear_all_tags">Clear all</string>
 
+    <string name="settings">Settings</string>
+    <string name="periodic_clearing">Periodic clearing</string>
+    <string name="enable_periodic_clearing">Periodic clearing service</string>
+    <string name="clearing_service_interval">Clearing service interval</string>
+    <string name="mock_interval_value">1 day</string>
```
Comment (michal1990) [inline app/src/main/res/values/strings.xml]: why did you create mock string resource in a production code?

File: app/src/main/res/layout/settings_activity.xml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            android:layout_height="wrap_content"
+            android:background="@color/colorPrimary"
+            android:theme="@style/ActionBarTheme"
+            app:popupTheme="@style/ThemeOverlay.AppCompat.Light"/>
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/text_view_margin"
+            android:text="@string/periodic_clearing"
```
Comment (kejmil01) [inline app/src/main/res/layout/settings_activity.xml]: I think it should be clearAccent color  (like in android settings).

File: app/src/main/res/layout/settings_activity.xml (hunk 5 above)
Comment (BartSoftware) [reply app/src/main/res/layout/settings_activity.xml]: Yes it doesn't work

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (BartSoftware) [reply app/src/main/res/values/strings.xml]: Its not a mock its just first value

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java (hunk 1 above)
Comment (BartSoftware) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: You mean to move startActivity(.. ,..) to separate method?

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java [hunk 8]
```diff
@@ -46,6 +50,25 @@ public void prepareView() {
         extractBundle();
     }
 
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.settings:
+                Intent intent = SettingsActivity.buildIntent(this);
+                startActivity(intent);
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: I would like to see these two lines extracted to method.

**Review by kejmil01 — APPROVED** [review]

**Review by michal1990 — APPROVED** [review]

**Pull request merged**